docs/geps/gep-06.md

```diff
@@ -0,0 +1,280 @@
+   refer to `anwartschaftszeit` directly. This will allow keeping code short,
+   self-contained, and clear.
+
+1. A current example for functions changing over the years would be
+   `midijob_bemessungsentgelt_m`. The relevant code in `policy_environment` is:
+
+   ```python
+   from _gettsim.social_insurance_contributions.eink_grenzen import (
+       midijob_bemessungsentgelt_m_ab_10_2022,
+   )
+   from _gettsim.social_insurance_contributions.eink_grenzen import (
+       midijob_bemessungsentgelt_m_bis_09_2022,
+   )
+
+   [...]
+
+   if date >= datetime.date(year=2022, month=10, day=1):
+       functions["midijob_bemessungsentgelt_m"] = midijob_bemessungsentgelt_m_ab_10_2022
+   else:
+       functions["midijob_bemessungsentgelt_m"] = midijob_bemessungsentgelt_m_bis_09_2022
+   ```
+
+   Adding a function that changes over time always means making changes in two
+   completely unrelated places, it is very easy to get this wrong.
+
+   A future implementation may look something like:
+
+   ```py
+   @dates_active(start="2003-04-01", end="2022-09-30", removesuffix="_some_spec")
+   def midijob_bemessungsentgelt_m_some_spec():
+       pass
+
+   @dates_active(start="2022-10-01", removesuffix="_some_spec")
+   def midijob_bemessungsentgelt_m_another_spec():
+       pass
+   ```
+
+   If `@dates_active` evaluates to `False`, the function will not appear in the policy
+   functions dictionary. Else, whatever is left after removing `removesuffix` will be
+   placed as a key in the functions dictionary.
+
+1. The yaml-files with parameters
+
+   - Will live in the same directory as the other functions (=same namespace).
+
+   - Will be parsed to dictionaries and that's it.
+
+   - Might be used in a similar way as they are now (argument `params`, use
+     `params["beitragssatz"]` to access elements) or get special treatment as arguments,
+     e.g., `p_beitragssatz`.
+
+     In both cases, the namespace makes clear we are talking about, say, `arbeitsl_v`.
+     If we need parameters which are external to the current namespace, we will need the
+     same verbose syntax as in 1. (`ges_rentenv___params` /
+     `ges_rentenv___params___beitragssatz`).
+
+     What we do will depend in part on how strict we will want to be about type
+     checking; mypy does like very general dictionaries. But in general, it can be
+     useful to see directly from the interface of a function what is being used inside.
+
+## Backward compatibility
+
+This section describes the ways in which the GEP breaks backward compatibility.
+
+The [#GEPs] stream will contain the GEP up to and including this section. This is to
+avoid losing users who are not interested in implementation details and instead focus
+the discussion on usage and impact of the intended features.
+
+The `compute_taxes_and_transfers` function will get an additional optional keyword
+argument, `columns_mapping`. As the name suggests, it maps columns in the user-provided
```

Comment:
We should give an example of `columns_mapping`. For me it is not entirely clear.

Reply:
See c1a48f4